frame/dapps-staking/src/lib.rs

```diff
@@ -58,10 +59,12 @@ impl Default for Forcing {
 
 /// A record for total rewards and total amount staked for an era
 #[derive(PartialEq, Eq, Clone, Default, Encode, Decode, RuntimeDebug, TypeInfo)]
-pub struct EraRewardAndStake<Balance> {
+pub struct EraRewardAndStake<Balance: HasCompact> {
     /// Total amount of rewards for an era
+    #[codec(compact)]
```

Comment:
This isn't backward compatible change, do you have storage migration?

Reply:
Yes, here: https://github.com/PlasmNetwork/Astar/pull/509/files#diff-3025024e1e1f1f1d09b141476522527d9c45bf605315fe06dbe89d57cc9daf73R83-R90